weave/type_wrappers/Content/content.py

```diff
@@ -0,0 +1,250 @@
+    input: str | bytes, /, **values: Unpack[ContentArgs]
+) -> BaseContentHandler:
+    try:
+        data = base64.b64decode(input, validate=True)
+        values["extra"]["input_type"] = values["extra"].get("input_type") or str(
+            type(input)
+        )
+        values["extra"]["input_category"] = (
+            values["extra"].get("input_category") or "base64"
+        )
+        return create_bytes_content(data, **values)
+    except Exception as e:
+        raise ValueError(f"Invalid base64 string: {e}") from e
+
+
+T = TypeVar("T", bound=str)
+
+
+class Content(Generic[T]):
+    """
+    This is similar to a factory but uses the instance as an attribute so we can
+    benefit from the organization of the content handlers without having to
+    manage the object recognition for the weave serializers
+    """
+
+    content_handler: BaseContentHandler
+    _last_saved_path: str | None = None
+
+    def __init__(
+        self,
+        input: Any,
+        type_hint: str | None = None,
+        /,
+        **values: Unpack[ContentKeywordArgs],
+    ):
+        if type_hint:
+            if type_hint.find("/") != -1:
+                values["mimetype"] = type_hint
+            else:
+                values["extension"] = type_hint.lstrip(".")
+
+        extra = values.get("extra", {})
+
+        content_args: ContentArgs = {
+            "filename": values.get("filename", None),
+            "path": values.get("path", None),
+            "extension": values.get("extension", None),
+            "mimetype": values.get("mimetype", None),
+            "size": values.get("size", None),
+            "extra": extra,
+        }
+
+        if isinstance(input, Path):
+            content_args["extra"]["input_type"] = content_args["extra"].get(
+                "input_type"
+            ) or str(input.__class__)
+            content_args["extra"]["input_category"] = (
+                content_args["extra"].get("input_category") or "object"
+            )
+            self.content_handler = create_file_content(str(input), **content_args)
+
+        elif isinstance(input, str):
+            if is_valid_path(input):
+                self.content_handler = create_file_content(str(input), **content_args)
+            else:
+                try:
+                    self.content_handler = create_b64_content(
+                        str(input), **content_args
+                    )
+                except ValueError:
+                    raise ValueError(
+                        f"Could not parse string {input} as a valid path or base64 string"
+                    )
+        elif isinstance(input, bytes):
+            self.content_handler = create_bytes_content(input, **content_args)
+        else:
+            raise TypeError(f"Unsupported input type: {type(input)}")
+
+    @property
+    def metadata(self) -> dict[str, Any]:
+        return self.content_handler.metadata
+
+    @property
+    def input_type(self) -> str | None:
```

Comment:
If there's anything in this PR that I would guess is up for debate it's how `input_type` and `input_category` are being handled. I made a design choice here after trying it a few different ways and finding this one worked best, but I'm open to input.